Source/Scene/WebMapServiceImageryProvider.js

```diff
@@ -208,27 +210,12 @@ define([
 
     function requestImage(imageryProvider, col, row, level, request, interval) {
         var dynamicIntervalData = defined(interval) ? interval.data : undefined;
-        var resource = imageryProvider._tileProvider._resource; // We actually want to set the time parameter within the tile provider.
-        var parameters = {};
-        var keys = [];
+        var resource = imageryProvider._tileProvider._resource; // We actually want to set the query parameters within the tile provider.
```

Comment:
Since we can have multiple images in flight, including some for a future time with different dynamic data, you want something like this, that will clone the resource but merge in the dynamic query parameters. Also, you don't need to check if `dynamicIntervalData` is an object.

You want to do something like
```javascript
var resource = imageryProvider._tileProvider._resource; 
if (defined(dynamicIntervalData)) {
    resource = resource.getDerivedResource({
        queryParameters: dynamicIntervalData
    });
}
```

Reply:
The reason why I am updating the parameters on the resource in the tile provider is because it is the one that is using them.  If I make a copy of the resource as you suggest, then I have no way of setting that as the resource in the tile provider (in this case the UrlTemplateImageryProvider).  It doesn't feel right to update the parameters on the resource in the tile provider, but the API for requestImage does not allow for passing the parameters any other way.  Thoughts?